csi/node_server.go

```diff
@@ -474,29 +500,41 @@ func (ns *NodeServer) NodeStageVolume(ctx context.Context, req *csi.NodeStageVol
 }
 
 func (ns *NodeServer) NodeUnstageVolume(ctx context.Context, req *csi.NodeUnstageVolumeRequest) (*csi.NodeUnstageVolumeResponse, error) {
-	log := getLoggerForCSIControllerServer()
-	log = log.WithFields(logrus.Fields{"function": "NodeUnstageVolume"})
+	log := ns.log.WithFields(logrus.Fields{"function": "NodeUnstageVolume"})
 
-	targetPath := req.GetStagingTargetPath()
-	if targetPath == "" {
-		return nil, status.Error(codes.InvalidArgument, "target path missing in request")
+	stagingTargetPath := req.GetStagingTargetPath()
+	if stagingTargetPath == "" {
+		return nil, status.Error(codes.InvalidArgument, "staging target path missing in request")
 	}
 
 	volumeID := req.GetVolumeId()
 	if len(volumeID) == 0 {
 		return nil, status.Error(codes.InvalidArgument, "volume id missing in request")
 	}
 
+	mounter := mount.New("")
+
 	// CO owns the staging_path so we only unmount but not remove the path
-	if err := unmount(targetPath, mount.New("")); err != nil {
-		return nil, status.Error(codes.Internal, fmt.Sprintf("failed to unmount volume %s mount point %v error %v", volumeID, targetPath, err))
+	if err := unmount(stagingTargetPath, mounter); err != nil {
```

Comment:
If it's an encrypted block volume, do we still need to unmount `stagingTargetPath`? or just `getStageBlockVolumePath(stagingTargetPath, volumeID)` only?

Reply:
`unmount` is a no op for a block mode volume.
https://github.com/longhorn/longhorn-manager/blob/master/csi/util.go#L338